### DISCUSSION TITLE
Next js api not working on production but in local it's working.

### DISCUSSION THREAD

**Discussion by Babailan** [vercel/next.js#35808] [Help] [answered]
### I got 500 internal error on serverless function
it's working on local dev but when I upload it on vercel the api isn't working anymore

[test api](https://babailan-animecharacter.vercel.app/api/trailer)

[repository so you can view](https://github.com/Babailan/animeCharacterApp)

**Comment by icyJoseph** [comment, marked as answer]
Hi,

I wonder if downloading videos like that is considered `fair use` by Vercel, according to https://vercel.com/docs/concepts/limits/fair-use-policy, I have a feeling it might not be `fair use`. 

In such a case, I think you might have to pay.

It could also be that there's some system dependency missing, but I think the package you are using is 100% JS.

I'd try to put a console.log and check the Vercel Function logs, there has to be a better explanation there.

**Reply by Babailan** [reply]
Thanks alot

**Reply by icyJoseph** [reply]
Please do let us know if the vercel logs say something about the failure.

**Reply by Babailan** [reply]
@icyJoseph it says that I exceed The maximum size of responses is 4 MB. do you know how would i fix this Thanks.

**Reply by Babailan** [reply]
@icyJoseph I think that's not the problem

**Reply by icyJoseph** [reply]
Aha, I see, then this is the problem, it is over the functional limit for serverless functions, https://vercel.com/docs/concepts/limits/overview#serverless-function-payload-size-limit - I am not sure if you can do the `ytdl` thing through serverless then. See this recommendation https://vercel.com/support/articles/how-to-bypass-vercel-5mb-body-size-limit-serverless-functions.

**Reply by Babailan** [reply]
@icyJoseph thanks alot i think i should use cloudinary btw how can i get a pro budge like you have.

**Reply by icyJoseph** [reply]
Ah, sorry, I don't know about that. You'll have to look it up. Responding to whether or not Cloudinary required a credit card.

You can google about GitHub Pro accounts.

**Reply by deepthapliyal** [reply]
No check your url where you're fetching ```localhost:3000/api``` this won't work in production.

**Reply by humailsaleem123** [reply]
so what is the solution to run the apis in production ?

**Reply by deepthapliyal** [reply]
Just use your ```hosted url/api```

**Reply by Babailan** [reply]
@humailsaleem123 you can use vercel serverless api which is free and cost nothing. vercel also support nextjs api.

**Reply by humailsaleem123** [reply]
@Babailan bro, my apis are working perfectly fine in dev mode but when its come to production I run locally, npm run build command but there is an error 
 info Collecting page data .ERRORS AxiosError: connect ECONNREFUSED ::1:3000
    at AxiosError.from (file:///E:/HUMAIL/NEXT.JS%20FOLDER/ROAD%20TO%20DEVELOPERS/road-to-developers/node_modules/axios/lib/core/AxiosError.js:89:14)
    
> Build error occurred
Error: Failed to collect page data for /roadmap/[roadmap_name]

can you please tell how to resolve this ?

**Reply by Babailan** [reply]
It's possible that the specified port is currently used by another process. sharing additional details could enable me to assist you better. If you're open to it, you might consider hosting your project on GitHub.
